modules/apps/segments/segments-context-vocabulary-test/src/testIntegration/java/com/liferay/segments/context/vocabulary/internal/persistence/listener/test/SegmentsContextVocabularyConfigurationModelListenerTest.java

```diff
@@ -0,0 +1,137 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.segments.context.vocabulary.internal.persistence.listener.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.kernel.util.LocaleThreadLocal;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.Dictionary;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Yurena Cabrera
+ */
+@RunWith(Arquillian.class)
+public class SegmentsContextVocabularyConfigurationModelListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final TestRule testRule = new LiferayIntegrationTestRule();
+
+	@Test
+	public void testAddVocabularyConfig() throws Exception {
+		String entityFieldName = RandomTestUtil.randomString();
+		String assetVocabularyName = RandomTestUtil.randomString();
+
+		Dictionary<String, Object> properties =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"assetVocabularyName", assetVocabularyName
+			).put(
+				"companyId", "987"
+			).put(
+				"entityFieldName", entityFieldName
+			).build();
+
+		Configuration configuration =
+			_configurationAdmin.createFactoryConfiguration(
+				_SEGMENTS_CONTEXT_VOCABULARY_CONFIGURATION_PID,
+				StringPool.QUESTION);
+
+		try {
+			configuration.update(properties);
+		}
+		catch (Exception exception) {
+			Assert.fail();
+		}
+		finally {
+			properties = configuration.getProperties();
+
+			Assert.assertEquals(
+				entityFieldName, properties.get("entityFieldName"));
+			Assert.assertEquals("987", properties.get("companyId"));
+			Assert.assertEquals(
+				assetVocabularyName, properties.get("assetVocabularyName"));
+
+			configuration.delete();
+		}
+	}
+
+	@Test
+	public void testDuplicatedContextVocabularyConfigShouldThrowAnException()
+		throws Exception {
+
+		String entityFieldName = RandomTestUtil.randomString();
+		String assetVocabularyName = RandomTestUtil.randomString();
+
+		Dictionary<String, Object> properties =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"assetVocabularyName", assetVocabularyName
+			).put(
+				"companyId", "987"
+			).put(
+				"entityFieldName", entityFieldName
+			).build();
+
+		Configuration configuration =
+			_configurationAdmin.createFactoryConfiguration(
+				_SEGMENTS_CONTEXT_VOCABULARY_CONFIGURATION_PID,
+				StringPool.QUESTION);
+
+		LocaleThreadLocal.setThemeDisplayLocale(LocaleUtil.US);
+
+		try {
+			configuration.update(properties);
```

Comment:
Update should not fail  if you run it twice, why it is failing?

Reply:
It fails because we are updating with the same properties twice, so the second time we are duplicating

Reply:
Yeahh, we can update the same property as many times as we want, the add should fail, but the update it should not fail.

Reply:
Currently, that update is working more as an add than as an update that "overwrites" the whole configuration, there is no add operation available per se (As I can see in the store method in ConfigurationPersistenceManager.java)
 
 Please, correct me if I am wrong

Reply:
Hey @cgoncas,
The behaviour described is unexpected as update the same data at the same scoped should be allowed (video attached).
I'll create a bug

https://user-images.githubusercontent.com/85171101/222714908-feb6d11a-e8c8-4599-b6b6-e3d82335367a.mov

Reply:
Great!!!

In the mean time, what do you think if we send just the first test and once we solve the bug we add the proper test for the bug fix?

Reply:
I can make this test work swith a duplication, but not using exactly the same properties. 
It will be great having it as well for when we address the bug found.

Once is done I will forward the PR if you agree

Thanks!

Reply:
Totally agree.